src/test/java/org/cryptomator/cryptofs/CryptoFileChannelWriteReadIntegrationTest.java

```diff
@@ -175,6 +174,54 @@ public void afterEach() throws IOException {
 			Files.deleteIfExists(file);
 		}
 
+		//https://github.com/cryptomator/cryptofs/issues/173
+		@Test
+		public void testWritingToPreviouslyReadRegionIsSavedCorrectly() throws IOException {
+			int x = 16_384; //half of cleartext chunk size
+			try (var reader = FileChannel.open(file, CREATE, READ, WRITE)) {
+				try (var writer = FileChannel.open(file, CREATE, WRITE)) {
+					assertWritten(writer, x, x + 1); //write to chunk 0 and 1
+					assertRead(reader, 2 * x, 1); //read 1 byte of chunk 1
+					assertWritten(writer, 2 * x, 2 * x); //write to chunk 1
+				}
+			}
+
+			try (var reader2 = FileChannel.open(file, CREATE, READ)) {
+				assertRead(reader2, 2 * x, x); //read chunk 1
+			}
+
+		}

```

Comment:
Is the `read` in between the two writes of any relevance?

Imo the helper functions hurt readability (and aren't used elsewhere in this test):

```suggestion
		@Test
		public void testWritingToPreviouslyReadRegionIsSavedCorrectly() throws IOException {
			int halfChunks = 16_384;
			try (var writer = FileChannel.open(file, CREATE_NEW, WRITE)) {
				writer.writer(ByteBuffer.allocate(3 * halfChunks), 0); // write to chunk 0 and 1
				writer.writer(ByteBuffer.allocate(5 * halfChunks), 0); // write to chunk 0 - 2
			}

			try (var reader = FileChannel.open(file, READ)) {
				// assertAll for reada from chunks 0, 1, 2. 0+2 should be fine even before the fix
			}

		}
```

Reply:
Changed in https://github.com/cryptomator/cryptofs/pull/174/commits/3e6fe77e2b047502a198aabfa535d2fab637985e